### PR TITLE
Add ability to withdraw and invalidate consents

### PR DESCRIPTION
**Pull request by thomasleese**
This adds functionality allowing nurses to withdraw consent (marking consent that's already given and change to refused) or to invalidate consent (consent that's either refused or given and mark it as invalid so it's ignored).

## Screenshots

Screenshot 2024-11-06 at 17 48 54
Screenshot 2024-11-06 at 17 48 51
Screenshot 2024-11-06 at 17 47 55
Screenshot 2024-11-06 at 17 23 36
Screenshot 2024-11-06 at 16 53 42

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -343,7 +343,7 @@ en:
               blank: Choose who you are trying to get consent from
             reason_for_refusal:
               inclusion: Choose a reason
-            reason_for_refusal_notes:
+            notes:
               blank: Enter details for refusing
```
Comment (benilovj) [inline config/locales/en.yml:347]: I think the `blank` error messages need to be more general now?

Comment (thomasleese) [reply config/locales/en.yml]: Good point, I've amended the commit to go with "Enter notes" which I think matches our other error messages.

File: spec/features/invalidate_consent_spec.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    then_i_see_the_consent
+
+    when_i_click_on_the_consent
+    then_i_see_the_consent
+    and_i_click_invalidate_consent
+
+    when_i_fill_in_the_notes
+    and_i_click_invalidate_consent
+    then_i_see_the_consent_has_been_withdrawn
+    and_i_cant_withdraw
```
Comment (benilovj) [inline spec/features/invalidate_consent_spec.rb:21]: worth asserting here that patient session status has changed and that it's not possible to vaccinate anymore

File: app/components/app_activity_log_component.rb [hunk 3]
```diff
@@ -39,11 +39,40 @@ def all_events
   end
 
   def consent_events
```
Comment (benilovj) [inline app/components/app_activity_log_component.rb:41]: would be good add spec coverage for these events

**Comment by thomasleese** [comment]
@benilovj Thanks, I think all those points should be resolved now.

**Review by benilovj — APPROVED** [review]

**Pull request merged**